### ISSUE TITLE
Custom drawing 

### ISSUE THREAD

**Issue by orhanbalci** [Gekkio/imgui-rs#24]
Hi there;
As far as I understand it is not possible to do custom drawing with current wrapper. Do you plan to integrate custom drawing capabilities?

**Comment by Gekkio** [comment]
Do you mean custom manipulation of ImDrawList? Yes, this is not currently possible, but might be doable in the future. But I don't have any concrete plans at the moment, because I haven't used that API and have no idea how to wrap it safely.

**Comment by orhanbalci** [comment]
Actually I mean drawing using ImDrawList. For example 
[(https://github.com/ocornut/imgui/blob/master/imgui_demo.cpp#L333)]

**Comment by Gekkio** [comment]
Yes, that's what I mean. I think there's no technical restrictions, but the resulting API must be safe, so it's not necessarily easy to wrap it.

**Comment by orhanbalci** [comment]
Actually I was developing another library of my own and needed to write a GUI for debug purposes. Than I need custom drawing and realized the functionality is missing. If you have  a design on your mind I can give it a try.

**Comment by Gekkio** [comment]
Sorry, I haven't studied the API so I don't have any design ideas at the moment.
In general, a couple of things should be kept in mind:

* the safe wrapper should wrap all the things that make sense from a high-level/Rust point of view. It doesn't have to be 1 to 1, but should try to be complete
* the safe wrapper **must be safe**. This requires a lot of effort and good knowledge of Rust, C/C++, ImGui, and low-level semantics. I've tried really hard to keep the library crash-free, but I'm sure I've made some mistakes if you dig deep enough
* the safe wrapper should follow Rusty idioms whenever possible
